sdks/java/extensions/ml/src/test/java/org/apache/beam/sdk/extensions/ml/RecommendationAIUserEventIT.java

```diff
@@ -0,0 +1,104 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.ml;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.api.client.json.GenericJson;
+import com.google.cloud.recommendationengine.v1beta1.UserEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RecommendationAIUserEventIT {
+  @Rule public TestPipeline testPipeline = TestPipeline.create();
+
+  public static GenericJson getUserEvent() {
+    GenericJson userInfo = new GenericJson().set("visitorId", "1");
+    return new GenericJson().set("eventType", "page-visit").set("userInfo", userInfo);
+  }
+
+  @Test
+  public void createUserEvent() {
+    String projectId = testPipeline.getOptions().as(GcpOptions.class).getProject();
+
+    PCollectionTuple createUserEventResult =
+        testPipeline
+            .apply(
+                Create.of(Arrays.asList(getUserEvent()))
+                    .withCoder(GenericJsonCoder.of(GenericJson.class)))
+            .apply(RecommendationAIWriteUserEvent.newBuilder().setProjectId(projectId).build());
+    PAssert.that(createUserEventResult.get(RecommendationAIWriteUserEvent.SUCCESS_TAG))
+        .satisfies(new VerifyUserEventResult(1));
+    testPipeline.run().waitUntilFinish();
+  }
+
+  @Ignore("Import method causing issues")
+  @Test
+  public void importUserEvents() {
+    String projectId = testPipeline.getOptions().as(GcpOptions.class).getProject();
+    ArrayList<KV<String, GenericJson>> userEvents = new ArrayList<>();
+    userEvents.add(KV.of("123", getUserEvent()));
+    userEvents.add(KV.of("123", getUserEvent()));
+
+    PCollectionTuple importUserEventResult =
+        testPipeline
+            .apply(Create.of(userEvents))
+            .apply(RecommendationAIImportUserEvents.newBuilder().setProjectId(projectId).build());
+    PAssert.that(importUserEventResult.get(RecommendationAIWriteUserEvent.SUCCESS_TAG))
+        .satisfies(new VerifyUserEventResult(2));
+    testPipeline.run().waitUntilFinish();
+  }
```

Comment:
What's the difference between Import and Write user events?

Reply:
Good point. The API was returning faulty responses (hence the test is currently Ignored) - it would require inspecting the `OperationFuture`-object returned from the API which should return the original resources as per the [docs](https://cloud.google.com/recommendations-ai/docs/reference/rest/v1beta1/projects.locations.catalogs.eventStores.operations#Operation) (so measuring the size of the array returned should be the way to go here).

Should I remove the test for now or leave a comment with this information?